packages/terra-consumer-nav/src/components/nav-toggler/NavToggler.jsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import Toggler from 'terra-toggle';
+import Arrange from 'terra-arrange';
+import OutlineChevronDown from '../../icons/OutlineChevronDown';
+import OutlineChevronUp from '../../icons/OutlineChevronUp';
+import styles from './NavToggler.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  text: PropTypes.string,
+  subItems: PropTypes.arrayOf(
+    PropTypes.shape({
+      text: PropTypes.string,
+      uri: PropTypes.string,
+      isActive: PropTypes.bool.isRequired,
+    }),
+  ),
+  handleToggle: PropTypes.func.isRequired,
+  isCollapsed: PropTypes.bool,
+};
+
+const defaultProps = {
+  isCollapsed: true,
+};
+
+const NavToggler = ({
+  text,
+  subItems,
+  handleToggle,
+  isCollapsed,
+  ...customProps
+}) => {
+  const tabIndex = isCollapsed ? -1 : 0;
+
+  const subNavs = subItems.map((element) => {
```

Comment:
this component array should just be passed as a prop... don't do this logic in here.